rfcs/proposed/simplified_numa_support/increased_availability/README.org

```diff
@@ -0,0 +1,101 @@
+# -*- fill-column: 80; -*-
+
+#+title: Improve predictability of API for NUMA support
+
+*Note:* This is a sub-RFC of the https://github.com/oneapi-src/oneTBB/pull/1535.
+Specifically, its section about "Increased availability of NUMA support".
+
+* Introduction
+oneTBB has soft dependency on several versions of ~tbbbind~, which are loaded by
+the library as part of its initialization stage. In turn, each ~tbbbind~ has
+hard dependency, i.e., relies on load-time linking, on concrete version of the
+HWLOC library [1, 2]. The soft dependency of oneTBB on ~tbbbind~ allows the
+library to continue its execution even if system loader is unable to resolve
+hard dependency on HWLOC for ~tbbbind~. In this case, the NUMA support API does
+nothing. An error is also not reported and the code that uses NUMA support
+facilities may continue running expecting it to work. Such behavior is not
+readily noticable by the users of oneTBB and this represents the main problem
+with the current behavior.
+
+Having a dependency on a shared HWLOC library has a number of advantages:
+1. Code reuse with all of the positive consequences out of this. That's the
+   primary reason shared libraries are for.
+2. Sharing HWLOC context between its clients. This avoids performing multiple
+   times same operations issuing identical results.
+3. A drop-in replacement. Users are able to use their own version of HWLOC
+   without recompilation of oneTBB.
```

Comment:
Since in the context of this discussion "users" can be either developers that use oneTBB or application end users, I would recommend to make it clear each time - for example, change "users" to "developers" for the first category.